File: problems/greedy_algorithms/minimum_waiting_time/js/minimum_waiting_time.js
export const minimumWaitingTime = (queries) => {
  queries.sort((a, b) => a - b);

  let total = 0,
    query = 0;

  for (let i = 0; i < queries.length; i++) {
    total += query * (queries.length - i);
    query = queries[i];
  }
  return total;
};

File: problems/greedy_algorithms/minimum_waiting_time/cpp/minimum_waiting_time.cpp
#include "minimum_waiting_time.h"
#include <algorithm>

int minimumWaitingTime(std::vector<int> queries) {
  int total = 0, query = 0;
  std::sort(queries.begin(), queries.end());

  for (size_t i = 0; i < queries.size(); i++) {
    total += (query * (queries.size() - i));
    query = queries[i];
  }
  return total;
}
